streams/integration-tests/src/test/java/org/apache/kafka/streams/integration/KafkaStreamsTelemetryIntegrationTest.java

```diff
@@ -419,8 +485,36 @@ private Topology complexTopology() {
         return builder.build();
     }
 
-    private Topology simpleTopology() {
+    private void addGlobalStore(final StreamsBuilder builder) {
+        builder.addGlobalStore(Stores.keyValueStoreBuilder(
+                Stores.inMemoryKeyValueStore("iq-test-store"),
+                Serdes.String(),
+                Serdes.String()
+        ),
+                globalStoreTopic,
+                Consumed.with(Serdes.String(), Serdes.String()),
+                () -> new Processor<>() {
+                    private KeyValueStore<String, String> store;
+
+                    @Override
+                    public void init(final ProcessorContext<Void, Void> context) {
+                        store = context.getStateStore("iq-test-store");
+                    }
+
+                    @Override
+                    public void process(final Record<String, String> record) {
+                        store.put(record.key(), record.value());
+                        globalStoreIterator = store.all();
```

Comment:
The store iterator is intentionally not closed here as it needs to be open during the test so the Streams app will emit the `org.apache.kafka.stream.state.oldest.iterator.open.since.ms` metric that is expected.  So the `globalStoreIterator` is a global variable (pun not intended) so it can be closed in the `tearDown` method.

Reply:
Uff... quick a hack. But I guess difficult (impossible) to do differently?

We should add a comment about it though, ie, just c&p your PR comment directly (it's a good explanation, and avoids that somebody spends time tryin to "fix" it, just breaking the test and getting confused, and wasting time).

We might also want to add a `@SuppressWarning("resource")` to `process(...)` method? IntelliJ should complain about not using try-with-resource for the iterator?

Reply:
Regarding `@SuppressWarning` I wasn't seeing one ,so I didn't add it.  If you'd prefer to have on anyway I'll do so.

Reply:
Seems my IntelliJ warn setting are different. Should be ok w/o it, too.